myworld1/web_scrapper.py

```diff
@@ -0,0 +1,95 @@
+import psycopg2
+import requests
+import re
+from bs4 import BeautifulSoup, element
+import os
+        
+# For the credentials mentioned below, you may refer the docker-compose.yml present in myworld .
+db_name = 'member_db'
+db_user = 'postgres'
+db_pass = '123456'
+db_host = 'psql-db'
+db_port = '5432'
+        
+# This will create the connection the to postgres database.
+conn = psycopg2.connect(dbname=db_name, user=db_user, password=db_pass, host=db_host, port=db_port)
+                   
+                    
+def add_row_to_blog(title, author, date, time,content,recommended,path):
+    # This function will add the entry to database
+    sql = """INSERT INTO members_blog (title, release_date, blog_time, created_date,content, author,recommended,path) VALUES (%s, %s::DATE, %s::TIME,  NOW(),%s,%s,%s,%s)"""
+            
+    with conn:
```

Comment:
@AnkitaHegde Why `with` is used?

Reply:
Because **with** will perform automatic closing of resources. If not used we have explicitly close those connection and cursors.

Reply:
@AnkitaHegde Ok, so you mentioned about "closing of resources". I'll agree that it's correct. But you're actually inserting data into the table and in another method you're truncating the table, I don't see where you're committing the transaction. So how is the data stored in the database if there's no `commit`?

Reply:
By default database will be in auto-commit mode . So commit statement is not explicitly written.

Reply:
@AnkitaHegde Could you add the resource which says that the database will be in auto-commit mode by default or how did you conclude it?

Reply:
Well, its like the session starts with auto-commit mode. If we want to disable it only then we will be writing explicitly as autocommit=0.

Reply:
@AnkitaHegde Could you add the link to resource which says so?

Reply:
This is the link-
https://dev.mysql.com/doc/refman/8.0/en/innodb-autocommit-commit-rollback.html#:~:text=By%20default%2C%20connection%20to%20the,statement%20as%20you%20execute%20it.

Reply:
@AnkitaHegde which database are you using to store the data in your code?

Reply:
Its postgres. Here its about sql but I thought those auto-commit things are similar in these, aren't they?

Reply:
@AnkitaHegde The line quoted by you in that article clearly talks about mysql behaviour and not SQL in general. Also whatever mentioned in that paragraph(even in case of **postgres**) holds true for **console** where you're actually executing the SQL queries directly and it is auto-committed. 
But in your case, you're actually executing queries using your python script and you're using **psycopg2** for it. This means you're not directly interacting with the database but using a API(psycopg2) to interact. So my original question still stands.
So, I would want you to understand what is `with` and then how/why `with` is used?
You can comment your answer regarding what you've understood and then we can move on to answering the original question(provided your understanding of `with` is correct).

Reply:
@AnkitaHegde As I said before, first I want you understand `with` and why would someone use `with`, on your own and then comment out your understanding on it, so that I can correct you(if you're wrong at some point) or I can help you connect the dots.
You can write your own simple scripts and experiment with it. Take your time to understand it.